catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/XmlInputTransformer.java

```diff
@@ -244,4 +238,60 @@ public void setDescription(String description) {
     public void setOrganization(String organization) {
         this.organization = organization;
     }
+
+    /**
+     * Defines and returns a {@link DynamicMetacardType} based on component Sax Event Handler Factories
+     * and what attributes they populate
+     *
+     * @return a DynamicMetacardType that describes the type of metacard that is created in this transformer
+     */
+
+    public MetacardType getMetacardType() {
+        Set<AttributeDescriptor> attributeDescriptors = new HashSet<>();
+
+        if (saxEventHandlerConfiguration != null && saxEventHandlerFactories != null) {
+            List<SaxEventHandlerFactory> filteredSaxEventHandlerFactories =
+                    saxEventHandlerFactories.stream()
+                            .filter(p -> saxEventHandlerConfiguration.contains(p.getId()))
+                            .collect(Collectors.toList());
+
+            for (SaxEventHandlerFactory factory : filteredSaxEventHandlerFactories) {
+                attributeDescriptors.addAll(factory.getSupportedAttributeDescriptors());
+            }
+        }
+        attributeDescriptors.addAll(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
+        return new DynamicMetacardType(attributeDescriptors, id);
+    }
+
+}
+
+class DynamicMetacardType implements MetacardType {
+
+    Set<AttributeDescriptor> attributeDescriptors;
+
+    String name;
+
+    public DynamicMetacardType(Set<AttributeDescriptor> attDesc, String name) {
+        this.attributeDescriptors = attDesc;
+        this.name = name + ".metacard";
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public Set<AttributeDescriptor> getAttributeDescriptors() {
+        return attributeDescriptors;
+    }
+
+    @Override
+    public AttributeDescriptor getAttributeDescriptor(String attributeName) {
+        return attributeDescriptors.stream()
```

Comment:
There's no reason/value in creating the collection and then returning the first element. Change to
```java
return attributeDescriptors.stream()
        .filter(p -> p.getName()
                .equals(attributeName))
        .findFirst()
        .orElse(null);
```